### ISSUE TITLE
Partial Formatter with "US" default region breaks for phone numbers starting with "3"

### ISSUE THREAD

**Issue by katmarshgriffin** [marmelroy/PhoneNumberKit#714]
### New Issue Checklist

- [X] Updated PhoneNumberKit to the latest version
- [X] Phone number formatted correctly on [JavaScript version](https://htmlpreview.github.io/?https://github.com/google/libphonenumber/blob/master/javascript/i18n/phonenumbers/demo-compiled.html)
- [X] I searched for [existing GitHub issues](https://github.com/marmelroy/PhoneNumberKit)

### Steps to reproduce
Using PartialFormatter, set the default region to US, and start entering a phone number beginning with "3"

##### Expected result
Formats to "3"

##### Actual result
Formats to "(3"

### Environment
SPM

### Notes
- All other numbers appear to format correctly.
- Numbers beginning with "3" also format correctly when the country code is provided before the "3".
- Tested on the javascript version with the number "3362530000" and AsYouTypeFormatter Results are correct.
- I updated the testUSTollFreeNumber() unit test to start with a "3" instead of an "8". This caused the test to fail, same reasoning as expected result and actual result note. Adding updated test below

```    func testUSTollFreeNumber() {
        let partialFormatter = PartialFormatter(phoneNumberKit: phoneNumberKit, defaultRegion: "US")
        var testNumber = "3"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "3")
        testNumber = "33"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "33")
        testNumber = "336"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "336")
        testNumber = "3362"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "336-2")
        testNumber = "33625"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "336-25")
        testNumber = "336253"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "336-253")
        testNumber = "3362530"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "336-2530")
        testNumber = "33625300"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "(336) 253-00")
        testNumber = "336253000"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "(336) 253-000")
        testNumber = "3362530000"
        XCTAssertEqual(partialFormatter.formatPartial(testNumber), "(336) 253-0000")
    }
```

**Comment by Brandon-P** [comment]
Thanks @katherinemarsh for highlighting these repo steps. We are currently experiencing the same issue in our implementation. We found this issue after investigating a parsing issue that caused users to be unable to edit their profile information. Updating to the most current release would resolve our existing issue but exposes this issue, which would prevent a substantial portion of our userbase from logging in with their phone number.

An additional piece of information to share is that we found that verifying in simulator is not sufficient as the issue was not reproducible in simulator but reproduced when built to device.

**Comment by bguidolim** [comment]
Thank for reporting that. I can take a look at it, however only after 2 to 3 weeks since I'm don't have access to my computer in that period.
Or if you fix that and open a PR, I will be glad to review it.

**Comment by BrentMifsud** [comment]
seeing this as well.

**Comment by BrentMifsud** [comment]
seems the problem is numbers beginning with 3 are not matching this leading numbers regex: `[24-9]|3(?:[02-9]|1[1-9])` and getting the wrong number formatter.

My domain knowledge about phone number validation is quite limited, so I don't know if the problem is with the regex itself or not.

Another issue I have noticed is that if you continue typing out the phone number, it never receives the trailing `)` after the area code.

So you end up with a number formatted like so: `(3333333333`

**Comment by BrentMifsud** [comment]
I found a temporary workaround here.

You need to make sure you strip out all `(`, `)`, ` `, and `-` before calling formatPartial.

If the partial formatter picks up a number in the broken format: `(3` the partial formatter just returns the raw value you pass in.

So on our end, as the user types values, we strip out the applied formatting before requesting a new value from PhoneNumberKit.

**Comment by bguidolim** [comment]
Hey folks!
Do you mind check if the branch of PR #727 fixes the issue?
